sched: finalize_cancel_locked can early return

TaiJuWu opened the pull request:
The code `SYS_SLIST_FOR_EACH_CONTAINER_SAFE` just for remove work from `pending_cancels`. 
After removing work successfully, the function can return early. 
It is unnecessary to iterate continuously.

cfriedt commented on kernel/work.c:
```diff
@@ -141,6 +141,7 @@ static void finalize_cancel_locked(struct k_work *work)
 		if (wc->work == work) {
 			sys_slist_remove(&pending_cancels, prev, &wc->node);
 			k_sem_give(&wc->sem);
+			break;
 		} else {
 			prev = &wc->node;
 		}
```
```suggestion
			break;
 		}

 		prev = &wc->node;
```

cfriedt reviewed (commented):
LGTM - just a nit

TaiJuWu committed:
sched: finalize_cancel_locked can early return

The code `SYS_SLIST_FOR_EACH_CONTAINER_SAFE` just for remove work
from `pending_cancels`.
After removing work successfully, the function can return early.
It is unnecessary to iterate continuously.

Signed-off-by: TaiJu Wu <tjwu1217@gmail.com>

TaiJuWu commented:
This RP can be merged to main branch or it need extra process?

TaiJuWu commented:
Thanks @cfriedt  for merge.
